hack/lib/etcd.sh

```diff
@@ -68,8 +68,15 @@ kube::etcd::cleanup() {
 kube::etcd::install() {
   (
     cd "${KUBE_ROOT}/third_party"
-    curl -fsSL --retry 3 https://github.com/coreos/etcd/releases/download/v${ETCD_VERSION}/etcd-v${ETCD_VERSION}-linux-amd64.tar.gz | tar xzf -
-    ln -fns "etcd-v${ETCD_VERSION}-linux-amd64" etcd
+    uname=$(uname)
+    if [[ "${uname}" == "Darwin" ]]; then
+      curl -fsSLO --retry 3 https://github.com/coreos/etcd/releases/download/v${ETCD_VERSION}/etcd-v${ETCD_VERSION}-darwin-amd64.zip
```

Comment:
kinda frustrating that etcd releases zips for OSX and tarballs for linux. sigh.

a part of me wonders if it'd be worth making `etcd-${ETCD_VERSION}-darwin-amd64` a variable, since it gets repeated three times, though I don't feel too strongly about it. WDYT?

it'd probably be worth deleting the zip file after extracting. in the linux path we never save the tarfile.